src/components/structures/RoomStatusBar.tsx

```diff
@@ -142,6 +166,30 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
         dis.fire(Action.FocusSendMessageComposer);
     };
 
+    private onRefreshTimelineClick = async (): Promise<void> => {
+        this.setState({
+            isRefreshing: true,
+            refreshError: null,
+        });
+        try {
+            // Empty out the current timeline and re-request it
+            await this.props.room.refreshLiveTimeline();
```

Comment:
I believe there to be a sizeable flaw in the current js-sdk implementation upon which this is based.

`MatrixClient::getLatestTimeline` calls /messages and then on the latest event calls `MatrixClient::getEventTimeline` which if after calling `/context` finds the event to be a thread relation will create a Thread and return its `liveTimeline`. This could result in that timeline having newly synced events appended to it even if they're not part of the thread causing a sizeable regression.

Reply:
Addressing this in:

 - https://github.com/matrix-org/matrix-js-sdk/pull/2521
 - And the "Perfectly resolves timelines when refresh fails and then another refresh causes `getLatestTimeline()` finds a threaded event" Cypress test in this PR

Reply:
Related to [MSC3874: Filtering threads from the `/messages` endpoint](https://github.com/matrix-org/matrix-spec-proposals/pull/3874)